Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/Rpow/IntegralRepresentation.lean

```diff
@@ -0,0 +1,412 @@
+      _ = (t ^ p * x / t) / (t + x) := Eq.symm (div_div ((t : ℝ) ^ p * ↑x) (↑t) (↑t + ↑x))
+      _ = (t ^ p / t * x) / (t + x) := by field_simp
+      _ = t ^ p / t * x / (t + x) := by ring
+      _ = t ^ (p - 1) * x / (t + x) := by congr; exact (Real.rpow_sub_one ht' p).symm
+  case pos =>
+    simp only [mem_Ioo] at hp
+    have hp₁ : p ≠ 0 := Ne.symm (ne_of_lt hp.1)
+    have hp₂ : p - 1 ≠ 0 := by linarith
+    simp [rpowIntegrand₀₁, ht', hp₁, hp₂]
+
+lemma rpowIntegrand₀₁_eqOn_pow_div (hp : p ∈ Ioo 0 1) (hx : 0 ≤ x) :
+    Set.EqOn (rpowIntegrand₀₁ p · x) (fun t => t ^ (p - 1) * x / (t + x)) (Ioi 0) := by
+  intro t ht
+  simp [rpowIntegrand₀₁_eq_pow_div hp (le_of_lt ht) hx]
+
+lemma rpowIntegrand₀₁_apply_mul (hp : p ∈ Ioo 0 1) (ht : 0 ≤ t) (hx : 0 ≤ x) :
+    rpowIntegrand₀₁ p (x * t) x = (rpowIntegrand₀₁ p t 1) * x ^ (p - 1) := by
+  have hxt : 0 ≤ x * t := by positivity
+  rw [rpowIntegrand₀₁_eq_pow_div hp hxt hx, rpowIntegrand₀₁_eq_pow_div hp ht zero_le_one]
+  by_cases hx_zero : x = 0
+  case neg =>
+    calc _ = x ^ (p - 1) * (t ^ (p - 1) * (x / (x * t + x))) := by
+              rw [← mul_assoc, mul_div_assoc]
+              congr
+              exact Real.mul_rpow hx ht
+      _ = x ^ (p - 1) * (t ^ (p - 1) * 1 / (t + 1)) := by
+              congr 1
+              simp only [mul_div_assoc]
+              congr 1
+              have : x * t + x = x * (t + 1) := by ring
+              rw [this, div_mul_eq_div_mul_one_div, div_self hx_zero, one_mul]
+      _ = t ^ (p - 1) * 1 / (t + 1) * x ^ (p - 1) := by rw [mul_comm]
+  case pos =>
+    simp [hx_zero]
+    apply Or.inr
+    refine Real.zero_rpow ?_
+    rw [mem_Ioo] at hp
+    linarith
+
+lemma rpowIntegrand₀₁_apply_mul' (hp : p ∈ Ioo 0 1) (ht : 0 ≤ t) (hx : 0 ≤ x) :
+    rpowIntegrand₀₁ p (x * t) x * x  = (rpowIntegrand₀₁ p t 1) * x ^ p := by
+  simp only [rpowIntegrand₀₁_apply_mul hp ht hx, mul_assoc]
+  congr
+  by_cases hx_zero : x = 0
+  case neg =>
+    have : 0 < x := lt_of_le_of_ne hx fun a => hx_zero (id (Eq.symm a))
+    field_simp [Real.rpow_sub this]
+  case pos =>
+    have : p ≠ 0 := by rw [mem_Ioo] at hp; linarith
+    simp [hx_zero, Real.zero_rpow this]
+
+lemma rpowIntegrand₀₁_apply_mul_eqOn_Ici (hp : p ∈ Ioo 0 1) (hx : 0 ≤ x) :
+    EqOn (fun t => rpowIntegrand₀₁ p (x * t) x * x)
+      (fun t => (rpowIntegrand₀₁ p t 1) * x ^ p) (Ici 0) :=
+  fun _ ht => rpowIntegrand₀₁_apply_mul' hp ht hx
+
+lemma continuousOn_rpowIntegrand₀₁ (hp : p ∈ Ioo 0 1) (hx : 0 ≤ x) :
+    ContinuousOn (rpowIntegrand₀₁ p · x) (Ioi 0) := by
+  refine ContinuousOn.congr ?_ <| rpowIntegrand₀₁_eqOn_pow_div hp hx
+  refine ContinuousOn.mul ?_ ?_
+  · refine ContinuousOn.mul ?_ continuousOn_const
+    exact ContinuousOn.rpow_const (by fun_prop) fun t ht => Or.inl <| Ne.symm (ne_of_lt ht)
+  · refine ContinuousOn.inv₀ (by fun_prop) ?_
+    intro t ht
+    simp only [mem_Ioo] at *
+    have : 0 < t + x := add_pos_of_pos_of_nonneg ht hx
+    exact Ne.symm (ne_of_lt this)
```

Comment:
Just supplying the one goal and then calling `fun_prop` with a discharger works.
```suggestion
  have h₀ : ContinuousOn (· ^ (p - 1) : ℝ → ℝ) (Ioi 0) := .rpow_const (by fun_prop) <|
    fun t ht => .inl ht.ne' 
  fun_prop (disch := intros; simp_all; positivity)
```